webview-ui/src/components/chat/Announcement.tsx

```diff
@@ -74,36 +88,27 @@ const Announcement = ({ hideAnnouncement }: AnnouncementProps) => {
 	)
 }
 
-const XLink = () => (
-	<VSCodeLink
-		href="https://x.com/roocode"
-		onClick={(e) => {
-			e.preventDefault()
-			vscode.postMessage({ type: "openExternal", url: "https://x.com/roocode" })
-		}}>
-		X
-	</VSCodeLink>
-)
-
-const DiscordLink = () => (
+const SocialLink = ({ icon, label, href }: { icon: ReactNode; label: string; href: string }) => (
```

Comment:
ReactNode and line 104 have inconsistent typings.  We should use one or the other.